core/qbft/qbft.go

```diff
@@ -199,21 +199,22 @@ func Run[I any, V Value[V]](ctx context.Context, d Definition[I, V], t Transport
 
 	// trimBuffer drops all older round's buffered messages.
 	trimBuffer := func() {
-		var selected []Msg[I, V]
-		for _, msg := range buffer {
-			if msg.Round() >= round {
-				selected = append(selected, msg)
-			}
-		}
-		buffer = selected
-
-		dedup := make(map[dedupKey]bool)
-		for k := range dedupIn {
-			if k.Round >= round {
-				dedup[k] = true
-			}
-		}
-		dedupIn = dedup
+		// TODO(corver): Fix trimming.
+		// var selected []Msg[I, V]
+		// for _, msg := range buffer {
+		//	if msg.Round() >= round-1 {
+		//		selected = append(selected, msg)
+		//	}
+		// }
+		// buffer = selected
+		//
+		// dedup := make(map[dedupKey]bool)
+		// for k := range dedupIn {
+		//	if k.Round >= round {
+		//		dedup[k] = true
+		//	}
+		// }
+		// dedupIn = dedup
```

Comment:
what is wrong with this approach? I assume there would be cases where we would need message of rounds before `round-1`

Reply:
We sometimes loose some PREPAREs from previous rounds, then we cannot trigger UnponQuorumRoundChange singe Qrc isn't justified. I have a plan to fix this in PR.